base/gmp.jl

```diff
@@ -443,6 +443,9 @@ end
 ^(x::Integer, y::BigInt ) = bigint_pow(BigInt(x), y)
 ^(x::Bool   , y::BigInt ) = Base.power_by_squaring(x, y)
 
+# override default inlining of x^2 and x^3 etc.
+^{p}(x::BigInt, ::Type{Val{p}}) = x^Culong(p)
```

Comment:
this gives an InexactError for negative powers, such as `BigFloat(2) ^ -20` (which was amazingly not tested)